jax/lax/lax_parallel.py

```diff
@@ -257,8 +282,10 @@ def standard_pmap_primitive(name, multiple_results=False):
   return prim
 
 
-def _allreduce_split_axis_rule(prim, reducer, vals, which_mapped, axis_name):
+def _allreduce_split_axis_rule(prim, reducer, vals, which_mapped, axis_name,
+                               axis_index_groups):
   assert tuple(which_mapped) == (True,)
+  assert axis_index_groups is None
```

Comment:
Is this an assertion or rather an error? An assertion is best for representing "if this code wrong it's a jax implementation bug", but if this is an issue of validating the inputs in this case perhaps it's better to raise a ValueError or NotImplementedError. (I always think of this in terms of how `python -O` disables assertions.)